screen/contributors/src/main/java/com/ivy/contributors/IvyWalletRepositoryDataSource.kt

```diff
@@ -33,26 +38,61 @@ class IvyWalletRepositoryDataSource @Inject constructor(
         @SerialName("stargazers_count")
         val stars: Int,
         @SerialName("html_url")
-        val url: String
+        val url: String,
     )
 
     companion object {
         private const val CONTRIBUTORS_PER_PAGE = 100
+        private const val DISPLAY_ANONYMOUS_CONTRIBUTORS = true
+        private const val INITIAL_PAGE = 1
     }
 
-    suspend fun fetchContributors(): List<ContributorDto>? {
-        return try {
-            withContext(Dispatchers.IO) {
+    suspend fun fetchContributors(): Either<String, List<ContributorDto>> = either {
+        withContext(dispatchersProvider.io) {
+            try {
+                pagingSource().bind()
+            } catch (e: Exception) {
```

Comment:
This `try-catch` should be unnecessary? Why do we need it? The `pagingSource` should be already a total function and never throw. It should be just withContext() and call to the pagingSource

Reply:
Yes, I was unsure about that because of the `nextPage` function, which can return null results. But since null results come from the left side, this should not cause any issues. So you're right! I’ll write some tests

Reply:
Tests ftw!